### PR TITLE
(project_list) add command to list all projects

### PR DESCRIPTION
**Pull request by em-**
I've taken inspiration from `lab issue list` and wrote a `project` subcommand to list all the available projects.

It doesn't provide any useful filtering for the moment, but it was enough for my needs. :)

### CODE REVIEW

File: docs/lab_project_list.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## lab project list
```
Comment (zaquestion) [inline docs/lab_project_list.md]: The docs are actually Auto generated during release to keep them aligned to the latest released version. Can you remove this file?

File: cmd/project_list.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	lab "github.com/zaquestion/lab/internal/gitlab"
+)
+
+var projectListCmd = &cobra.Command{
+	Use:     "list [page]",
+	Aliases: []string{"ls"},
+	Short:   "List all projects",
+	Run: func(cmd *cobra.Command, args []string) {
+		page := 0
+		if len(args) == 1 {
```
Comment (zaquestion) [inline cmd/project_list.go]: Can probably use the `parseArgs` function defined in `cmd/root.go` to accept a search and/or page. Similar to how other commands accept a remote and/or page.

Comment (zaquestion) [reply cmd/project_list.go]: I beleive ListProjectOptions can accept such a search parameter, but I may be mistaken.

File: cmd/project_list.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	"log"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/xanzy/go-gitlab"
+	lab "github.com/zaquestion/lab/internal/gitlab"
+)
+
+var projectListCmd = &cobra.Command{
```
Comment (zaquestion) [inline cmd/project_list.go]: Would like to see some sort of test, there are many examples in the `cmd` package.

File: cmd/project_list.go (hunk 2 above)
Comment (zaquestion) [reply cmd/project_list.go]: This is optional, was just thinking out loud here.

**Comment by zaquestion** [comment]
@em- I'm closing this out in favor of #173 I've picked up your changes (commit intact) and added search functionality + a test.

**Comment by em-** [comment]
Thank you so much!

**Pull request closed without merging**